codesign/bin/codesign.dart

```diff
@@ -71,14 +73,15 @@ Future<void> main(List<String> args) async {
       help: 'the Flutter engine commit revision for which Google Cloud Storage binary artifacts should be codesigned.',
     )
     ..addMultiOption(
-      kCodesignFilepathOption,
-      help: 'The zip file paths to be codesigned. Pass this option multiple'
-          'times to codesign multiple zip files',
-      valueHelp: 'darwin-x64/font-subset.zip',
+      kOptionalSwitch,
+      help:
+          'the list of binaries that are supported besides flutter engine binaries. e.g. ios-deploy.zip, libimobiledevice.zip, libplist.zip, libusbmuxd.zip, openssl.zip',
+      allowed: ["ios-deploy.zip", "libimobiledevice.zip", "libplist.zip", "libusbmuxd.zip", "openssl.zip"],
+      defaultsTo: <String>[],
     )
     ..addFlag(
-      kProductionFlag,
-      help: 'whether we are going to upload the artifacts back to GCS for production',
+      kDryrunFlag,
+      help: 'whether we are going to upload the artifacts back to GCS for dryrun',
```

Comment:
Dryrun seems to do a few things here. I would also expect it to skip the codesigning process. A short description may be "If passed, production services will not be used"

Reply:
umm yeah right currently `dryrun` signs the artifacts, notarizes them, but skips uploading them back to gcs. I guess the benefit of having the codesigning process is that notarization could potentially catch some extra exceptions, and the advantage of skipping codesigning is that we would have a faster dry run. Shall I change the code to make `dryrun` to skip codesigning as well? (If so I will probably do it in a follow up commit, planning to log off now to get dinner)

Reply:
Right in the newest commit I have changed `dryrun` to skip codesign and notarization.